client/src/components/ResourceList/ResourceList.test.js

```diff
@@ -1,47 +1,184 @@
-import { render, screen } from "@testing-library/react";
+import { render, screen, waitFor } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import { rest } from "msw";
+import React from "react";
+import { MemoryRouter } from "react-router-dom";
 
-import { resourceStub } from "../../../setupTests";
+import { resourceStub, server } from "../../../setupTests";
+import * as useFetchPublishedResourcesModule from "../../hooks";
+
+import TopicSelector from "./TopicSelector";
 
 import ResourceList from "./index";
 
+server.use(
+	rest.get("/api/topics", (req, res, ctx) => {
+		return res(
+			ctx.json([
+				{
+					id: "199a529b-22c1-460f-bc51-387cb12225e8",
+					name: "Git",
+				},
+				{
+					id: "84b099a4-8acd-4659-b5bd-1b89796fb924",
+					name: "HTML/CSS",
+				},
+			])
+		);
+	})
+);
+
+jest.mock("../../hooks/useFetchPublishedResources", () => ({
+	useFetchPublishedResources: jest.fn(),
+}));
+
+jest.mock("../../hooks/useFetchTopics", () => ({
+	useFetchTopics: jest.fn(),
+}));
```

Comment:
> Rather than mock the hooks, which don't actually have their own tests and therefore aren't covered _anywhere_, why not mock the requests as you did with `/api/topics` above (which isn't actually used due to the mocking of `useFetchTopics`)?

yes, that can be another more reliable option thank you.